internal/servers/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -614,3 +639,31 @@ func validateAuthenticateRequest(req *pbs.AuthenticateRequest) error {
 	}
 	return nil
 }
+
+func validateAuthenticateLoginRequest(req *pbs.AuthenticateLoginRequest) error {
+	badFields := make(map[string]string)
+	if strings.TrimSpace(req.GetAuthMethodId()) == "" {
+		badFields["auth_method_id"] = "This is a required field."
+	} else if !handlers.ValidId(password.AuthMethodPrefix, req.GetAuthMethodId()) {
+		badFields["auth_method_id"] = "Invalid formatted identifier."
+	}
+	// TODO: Update this when we enable different auth method types.
```

Comment:
If each auth method type uses a different subaction this TODO is no longer relevant so you can remove it.

Reply:
Done.